src/main.go

```diff
@@ -155,73 +175,139 @@ func realMain(ctx context.Context) error {
+			subtitle:        fmt.Sprintf("Release name: <b>%s</b>", getMapFieldStringValue(releaseContent, "name")),
+			ref:             getMapFieldStringValue(ghJSON, githubContextRefKey),
+			triggeringActor: getMapFieldStringValue(ghJSON, githubContextTriggeringActorKey),
+			timestamp:       getMapFieldStringValue(releaseContent, githubEventContenntCreatedAtKey),
+			clickURL:        getMapFieldStringValue(releaseContent, githubContextEventURLKey),
+			eventName:       "release",
+			repo:            getMapFieldStringValue(ghJSON, githubContextRepositoryKey),
+			headerIconURL:   successHeaderIconURL,
+		}
 	default:
-		iconUrl = "https://github.githubassets.com/favicons/favicon-failure.png"
+		res := &messageBodyContent{
+			title:           fmt.Sprintf("GitHub workflow %s", getMapFieldStringValue(jobJSON, "status")),
+			subtitle:        fmt.Sprintf("Workflow: <b>%s</b>", getMapFieldStringValue(ghJSON, "workflow")),
+			ref:             getMapFieldStringValue(ghJSON, githubContextRefKey),
+			triggeringActor: getMapFieldStringValue(ghJSON, githubContextTriggeringActorKey),
+			// The key for getting timestamp is different in differnet triggering event
+			// a simple work around is using the new timestamp.
+			timestamp: currentTimeStamp.UTC().Format(time.RFC3339),
+			clickURL:  fmt.Sprintf("https://github.com/%s/actions/runs/%s", getMapFieldStringValue(ghJSON, githubContextRepositoryKey), getMapFieldStringValue(ghJSON, "run_id")),
+			eventName: "workflow",
+			repo:      getMapFieldStringValue(ghJSON, githubContextRepositoryKey),
+		}
+		v, ok := jobJSON["status"]
+		if !ok || v == "failure" || v == "canceled" {
+			res.headerIconURL = failureHeaderIconURL
+		} else {
+			res.headerIconURL = successHeaderIconURL
+		}
+		return res
 	}
+}
 
+// generateRequestBody returns the body of the request.
+func generateRequestBody(m *messageBodyContent) ([]byte, error) {
 	jsonData := map[string]any{
 		"cardsV2": map[string]any{
 			"cardId": "createCardMessage",
 			"card": map[string]any{
 				"header": map[string]any{
-					"title":    fmt.Sprintf("GitHub workflow %s", jobJson["status"]),
-					"subtitle": fmt.Sprintf("Workflow: <b>%s</b>", ghJson["workflow"]),
-					"imageUrl": iconUrl,
+					"title":    m.title,
+					"subtitle": m.subtitle,
+					"imageUrl": m.headerIconURL,
 				},
 				"sections": []any{
 					map[string]any{
-						// "header":                    "This is the section header",
 						"collapsible":               true,
 						"uncollapsibleWidgetsCount": 1,
 						"widgets": []map[string]any{
 							{
 								"decoratedText": map[string]any{
 									"startIcon": map[string]any{
-										"iconUrl": "https://fonts.gstatic.com/s/i/short-term/release/googlesymbols/quick_reference/default/48px.svg",
+										"iconUrl": widgetRefIconURL,
 									},
-									"text": fmt.Sprintf("<b>Ref:</b> %s", ghJson["ref"]),
+									"text": fmt.Sprintf("<b>Repo: </b> %s", m.repo),
 								},
 							},
```

Comment:
Content in here is the same structure with different values, will have a refactor later to simplify this.